RFC - 16817 - AR standalone and beneficial ownership registrations (#16915)
bcgov/entity: rfcs/rfc-standalone-annual-report.md

# Summary

> Proposal:  create a standalone Annual Report filing that will contain the first instance of  the proposed "Register a beneficial owner" workflow. 

Reviewer: ```suggestion
Create a standalone Annual Report filing and Beneficial Ownership registration to prove out the workflow and UX for capturing the data required for the Transparency Registry.
```

  
2. The annual report standalone filing is a better place to start for collecting beneficial ownership information. For example, it would require less development work initially than starting with incorporation.
  
3. The sooner we can stand up a working version of the Register workflow, the sooner we can get it in front of users in the dev environment for UX research purposes. This is crucial because the OCM component of Registering beneficial owners will be extensive and challenging.
  
4. The sooner we can get data into the beneficial owner database (even just sample data) the sooner we can extend SOLR to beneficial owner database and include it along with director search to create the integrated Person Search.

# Detailed design

## Include Add a Beneficial Owner workflow

> The AR standalone filing will contain the information in the current corporate annual report filing and will also include the workflow for adding beneficial owners. The workflow includes:

Reviewer: ```suggestion
The filing flow would be:
```

1. Add Person section (name and email)
2. Significant Individual Assessment section (evaluates user input and returns status)

> 3. Add Personal Details (DOB, address, SIN, citizenship etc.)

Reviewer: ```suggestion
The workflow is only relevant for BCA entities.
1. Annual Report Date (this will be save to _lear_ or _colin_ as required)
2. Significant Individual Assessment section
3. UXA to determine:
   a. Add Beneficial Owner Details: OR
   b. Choose from Beneficial Ownership profile
4. Possibly create a Beneficial Ownership profile

# Summary

> Create a standalone Annual Report filing and Beneficial Ownership registration to prove out the workflow and UX for capturing the data required for the Transparency Registry.

Reviewer 2: test

> # Motivation

Reviewer 2: It's legislated that they have to provide ben owner info in an annual report; it's not legislated they have to file at time of creation.  But that doesn't mean the annual report needs to move from the Filings UI.  What is the extra value of moving the AR from the Filings UI to the RTR repo?  Businesses will launch any other filings/company changes from their business dashboard, why are we breaking the pattern?  Could we keep it in the Filings UI or Edit UI and then link out to the RTR repo to complete Ben owner info and then come back to finish off AR?

AR output should not include any ben owner info.

Reviewer: Having AR as something that is directly linkable and that we can make available today.
If we leave it in FilingsUI, then only business that have been migrated to Lear can use it.

I'm only suggesting AR, as it is a really small filing (the date) and we could send that info to Colin (already built into ColinAPI) or to Lear fairly easily.

This way we could launch or move into some UX Research now, without having to wait for the BCA types to migrate over.

Reviewer 2: We can't use it until OIC passes is early 2025.  By that time all corporations will be on Lear.

Reviewer: Based upon recent comments, I had thought we could start gathering in a non-required way in early 2024. Same as LOTR gathered data for 4 years before becoming mandatory and enforceable.

Reviewer: I wasn't suggesting AR include and BenOwner info. As we don't want to combine any of that data into the business registry, it would also give us a nice way to test flows through the linked apps, much like we link between the homepage, dashboards and filings.

  
2. The annual report standalone filing is a better place to start for collecting beneficial ownership information. For example, it would require less development work initially than starting with incorporation.
  
3. The sooner we can stand up a working version of the Register workflow, the sooner we can get it in front of users in the dev environment for UX research purposes. This is crucial because the OCM component of Registering beneficial owners will be extensive and challenging.
  
4. The sooner we can get data into the beneficial owner database (even just sample data) the sooner we can extend SOLR to beneficial owner database and include it along with director search to create the integrated Person Search.

# Detailed design

## Include Add a Beneficial Owner workflow
The filing flow would be:
The workflow is only relevant for BCA entities.
1. Annual Report Date (this will be save to _lear_ or _colin_ as required)
2. Significant Individual Assessment section
3. UXA to determine:
   a. Add Beneficial Owner Details: OR
   b. Choose from Beneficial Ownership profile
4. Possibly create a Beneficial Ownership profile

> 

Reviewer: Here's a sequence diagram to add in:
````suggestion
```sequenceDiagram
    actor Business Manager
    Business Manager->>+Annual Report: Need to file a businesses Annual Report
    Annual Report->>+Beneficial Ownership: Need to update Beneficial Owners
    Beneficial Ownership-->>-Annual Report: Beneficial Owners updated
    alt Business in Colin
      Annual Report->>Colin: File the AR Date
    else Business in Lear
          Annual Report->>Lear: File the AR Date
    end
    Annual Report-->>-Business Manager: Done!
```
````

Reviewer 2: I think for COLIN, it would be a full filing entry that we'd need to add, with an event_ID that is in sequence with the rest of the COLIN database as well as the Last AR year field (which might be on the corporation table - and is used for calculating Not in good standing).

Reviewer: We do that today for corps info we sync back to Colin. So that's not a new item to build

## Wireframes:

[https://www.figma.com/file/C2ThYIhZSO44dYBVGp8xAj/Register-Beneficial-Owners?type=design&node-id=0-1&mode=design]

> ## Personal Information (PI) Sequestered

Reviewer 2: Right, and shouldn't be on any filing outputs that are available to the public.

Beneficial owner data collected that is personal information is to be sequestered from business information. 

## Timeframe
The RTR team estimates this as a medium size epic: one sprint to create the environment and two sprints to code the AR standalone filing, for a total of 3 sprints.

# Drawbacks
(There are tradeoffs to choosing any path. Attempt to identify them here.
Why should we *not* do this? Please consider:

- implementation cost, both in term of code size and complexity

- whether the proposed feature can be implemented in user space

- the impact on teaching people

- integration of this feature with other existing and planned features

> - cost of migrating existing applications (is it a breaking change?)

Reviewer 2: I'd like to see a cost benefit analysis on this.  How much development work to migrate the filing, outputs, etc.

Reviewer: This wouldn't migrate any of the outputs. It would change notification and the outgoing link to file the AR.
It's also something we heard that orgs like, a reminder and link directly to the AR filing.

# Alternatives
(instructions:
- What is the impact of not doing this?

> - What other approaches to Registering a BO can be considered?)

Reviewer 2: We build only one stand-alone register in the RTR space and link out to it from other fiings like the Annual Report, Incorp, Amalg, Continuation In, Restoration.  Not sure how it would work linking back and forth.  Particularly when Incorp, Amalg and Continuation Ins wouldn't have an incorporation number until the filing is complete.

Reviewer: They all have a temporary number, that we use to work with them today.
